src/util/filesystem.cxx

```diff
@@ -0,0 +1,134 @@
+#include <filesystem.hxx>
+#include <Settings.hxx>
+#include <LOG.hxx>
+#include <fstream>
+
+#include <SDL.h>
+
+namespace fs
+{
+
+std::string readFileAsString(const std::string &fileName, bool binaryMode)
+{
+#ifdef __ANDROID__
+  // on Android, files cannot be accessed directly without Java. SDL implements that for us, so we use SDL RWOps until we implemented JNI File Access ourselves
+  // binary mode is ignored for now, savegames can't be loaded
+
+  SDL_RWops *rw = SDL_RWFromFile(fileName.c_str(), "rb");
```

Comment:
Why are we always reading file as binary on Android? Isn't this supposed to be text according to your comment?

Reply:
funnily, the one thing that needs binary (savegames) doesn't work, but everything else is loaded as normal string... idk.
we shouldn't keep the SDL_RWOps implementation anyway, and it works.